cmd/manager/main.go

```diff
@@ -140,10 +143,27 @@ func main() {
 		}
 	}
 
+	stopChannel := signals.SetupSignalHandler()
+
+	go func() {
+		log.Info("Creating Tekton installation resources once cache is sync'd")
+		if !mgr.GetCache().WaitForCacheSync(stopChannel) {
+			log.Error(errors.New("timed out waiting for caches to sync"), "")
+			os.Exit(1)
+		}
+		// create or update all NSTemplateTiers on the cluster at startup
+		log.Info("Creating the Tekton installation resource")
+		tektonInstallationCR, err := tektoninstallation.Asset("toolchain.openshift.dev_v1alpha1_tektoninstallation_cr.yaml")
+		if err = toolchain.CreateFromYAML(mgr.GetScheme(), mgr.GetClient(), tektonInstallationCR); err != nil {
+			log.Error(err, "Failed to create the 'TektonInstallation' custom resource during startup")
```

Comment:
just asking - we don't want to stop here, right? When there is a problem with creating the CR, then we just log and continue trying the che one, correct?

Reply:
@MatousJobanek that's a good question: there will be an error if the creation failed (but not if the CR already existed). But what's the point with continuing if we don't have the `TektonInstallation` to bootstrap? Or should we trying with Che and in worst case, have only Che installed?

Reply:
>  But what's the point with continuing if we don't have the TektonInstallation to bootstrap? Or should we trying with Che and in worst case, have only Che installed?

This is something we should maybe discuss with Parag - if the operator should be all-or-nothing, or if it should at least try to install one operator. From my point of view, I would exit here and stop the installation - the user/developer can retry the installation. It would be also good to check how it behaves and what it looks like when the creation fails - in both cases when we kill the installation or when we continue. Is there any error visible to the user? Is the whole installation properly stopped? WDYT?

Reply:
let me open a new issue to track this discussion, ok?

Reply:
:+1: